### PR TITLE
include rbd command output in errors, simplify ifelse logic

### PR DESCRIPTION
**Pull request by krmayankk**
**What this PR does / why we need it**:
As part of debugging  https://github.com/kubernetes/kubernetes/issues/54263, i found some room for improvements

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes # part of https://github.com/kubernetes/kubernetes/issues/54263

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/volume/rbd/rbd_util.go [hunk 1]
```diff
@@ -250,13 +256,13 @@ func (util *RBDUtil) AttachDisk(b rbdMounter) (string, error) {
...
+		found, rbdOutput, err := util.rbdStatus(&b)
 		if err != nil {
-			return "", err
+			return "", fmt.Errorf("error: %v, rbd output: %v", err, rbdOutput)
 		}
 		if found {
-			glog.Info("rbd is still being used ", b.Image)
-			return "", fmt.Errorf("rbd %s is still being used", b.Image)
+			glog.Info("rbd Image %v is still being used ", b.Image)
+			return "", fmt.Errorf("rbd Image %v is still being used. rbd output: %v", b.Image, rbdOutput)
```
Comment (cofyc) [inline pkg/volume/rbd/rbd_util.go]: Use `glog.Infof` instead, and change `Image %v` to `image %s/%s` then add b.Pool in arguments.

About `<pool>/<image>` format, see comments below.

File: pkg/volume/rbd/rbd_util.go [hunk 2]
```diff
@@ -277,33 +283,34 @@ func (util *RBDUtil) AttachDisk(b rbdMounter) (string, error) {
...
+			glog.V(1).Infof("rbd: map error %v, rbd output: %s", err, string(output))
 		}
 		if err != nil {
-			return "", fmt.Errorf("rbd: map failed %v %s", err, string(output))
+			return "", fmt.Errorf("rbd: map failed %v, rbd output: %s", err, string(output))
 		}
 		devicePath, found = waitForPath(b.Pool, b.Image, 10)
 		if !found {
-			return "", errors.New("Could not map image: Timeout after 10s")
+			return "", fmt.Errorf("Could not map Pool: %v image: %v Timeout after 10s", b.Pool, b.Image)
```
Comment (cofyc) [inline pkg/volume/rbd/rbd_util.go]: Change `Pool: %v iamge: %v` to `image %s/%s:`.

We can simply use `<pool>/<image>` format in logs, which use less characters. See examples: https://github.com/krmayankk/kubernetes/blob/460d26bb934656a74098732a3f812d3ee0a40f1a/pkg/volume/rbd/rbd_util.go#L205

`<pool>/<image>` image format is also widely used in `rbd` command, e.g. `rbd map kube/testimage`.

**Comment by cofyc** [comment]
@krmayankk 

hi, I added some comments.

@rootfs 

hi, simplification of if-else logic looks ok to me, but I don't know outputting command outputs in logs conform logging convention or not.

p.s. I cannot be assigned ;-)

**Comment by krmayankk** [comment]
@cofyc @rootfs thanks for the review. I want to show the command output from commands likes rbd lock list or other commands, in the describe events of the pods. I think this will be super useful when debugging errors. For example rbd lock list shows the name of the lock which in turn tells you which node is holding the lock. In case of errors , showing them in the events would be very helpful.

**Comment by rootfs** [comment]
@krmayankk the output may not be as helpful now. @cofyc has implemented rbd attach/detach. rbd lock is no longer held here, function `rbdLock` is only used to clean up old orphaned locks.

**Comment by krmayankk** [comment]
@rootfs questions:-
- is the rbd attach/dettach remotely done from kube-controller-manager or from the node ? implemenation proposal would be helpful to look 
- do you want me to remove the output for now ? I think getting the output in the error case could be useful. Let me know  ?

**Comment by cofyc** [comment]
@krmayankk RBD Attach/Detach interfaces: https://github.com/kubernetes/kubernetes/pull/51608.

File: pkg/volume/rbd/rbd_util.go [hunk 3]
```diff
@@ -174,19 +173,26 @@ func (util *RBDUtil) rbdLock(b rbdMounter, lock bool) error {
...
-							args = append(args, secret_opt...)
-							cmd, err = b.exec.Run("rbd", args...)
-							glog.Infof("remove orphaned locker %s from client %s: err %v, output: %s", lockInfo[1], lockInfo[0], err, string(cmd))
-						}
+			used, rbdOutput, statusErr := util.rbdStatus(&b)
+			if statusErr != nil {
+				return fmt.Errorf("rbdStatus failed error %v, rbd output: %v", statusErr, rbdOutput)
+			}
+			if used {
+				// this image is already locked by a node other than this node
```
Comment (rootfs) [inline pkg/volume/rbd/rbd_util.go]: nit: locked -> used (and error message below). Lock probe has not happened here yet

Comment (krmayankk) [reply pkg/volume/rbd/rbd_util.go]: fixed

**Pull request merged**